### PR TITLE
[NU-1466] deployment service scenario command

### PR DESCRIPTION
**Pull request by arkadius**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/ScenarioCommand.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// - validations - it is quick part, the result will be displayed on deploy modal
+// - deployment on engine side - it is longer part, the result will be shown as a notification
+case class RunDeploymentCommand(
+    processId: ProcessIdWithName,
+    savepointPath: Option[String],
+    comment: Option[String],
+    user: LoggedUser
+) extends ScenarioCommand[Future[Option[ExternalDeploymentId]]]
+
+case class CustomActionCommand(
```
Comment (lciolecki) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/ScenarioCommand.scala:27]: Is it still custom action? Why not just ActionCommand?

File: engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/EmbeddedDeploymentManager.scala [hunk 2]
```diff
@@ -99,11 +99,11 @@ class EmbeddedDeploymentManager(
...
         }
-      case command: CancelDeploymentCommand => cancelDeployment(command)
-      case command: CancelScenarioCommand   => cancelScenario(command)
-      case command: TestScenarioCommand     => processTestActionCommand(command)
-      case _: StopDeploymentCommand | _: StopScenarioCommand | _: MakeScenarioSavepointCommand |
-          _: CustomActionCommand =>
+      case command: DMCancelDeploymentCommand => cancelDeployment(command)
+      case command: DMCancelScenarioCommand   => cancelScenario(command)
+      case command: DMTestScenarioCommand     => processTestActionCommand(command)
+      case _: DMStopDeploymentCommand | _: DMStopScenarioCommand | _: DMMakeScenarioSavepointCommand |
```
Comment (lciolecki) [inline engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/EmbeddedDeploymentManager.scala:105]: To be consistent: processTestScenario?

**Review by lciolecki — APPROVED** [review]
Only minor stuff.. :)

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/ScenarioCommand.scala (hunk 1 above)
Comment (arkadius) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/ScenarioCommand.scala:27]: As you can see, we still have built-in actions (run deployment, cancel scenario), so we need some word to distinguish between them.

File: engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/EmbeddedDeploymentManager.scala (hunk 2 above)
Comment (arkadius) [reply engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/EmbeddedDeploymentManager.scala:105]: I guess that you want to left the comment next to the line above?  Take a look that methods handling concrete command don't have neighter process* nor *command. I think that for sake of consistency it should be rather just `testScenario`. WDYT? I also changed `validate` -> `validateScenario` to be more explicit.

**Pull request merged**